core/runner/src/main/java/com/cognifide/aet/runner/SuiteExecutorService.java

```diff
@@ -78,14 +84,24 @@ public void deactivate() {
     scheduledSuites.clear();
   }
 
-  void scheduleSuite(Suite suite, Destination jmsReplyTo) {
-    LOGGER.debug("Scheduling {}!", suite);
-    final ListenableFuture<String> suiteExecutionTask = executor
-        .submit(new SuiteExecutionTask(suite, jmsReplyTo, suiteDataService, runnerConfiguration,
-            suiteExecutionFactory));
-    scheduledSuites.add(suite.getCorrelationId());
-    Futures.addCallback(suiteExecutionTask, new SuiteFinishedCallback(suite.getCorrelationId()),
-        callbackExecutor);
+  void scheduleSuite(Run objectToRun, Destination jmsReplyTo) {
+    LOGGER.debug("Scheduling {}!", objectToRun.getObjectToRun());
+    final ListenableFuture<String> suiteExecutionTask;
+
+    Map<RunType, ProcessorStrategy> strategiesMap = new HashMap<>();
```

Comment:
You could create `ProcessingStrategyFactory` and move this logic to it (see pseudocode below).

```java
class ProcessingStrategyFactory {
   ProcessorStrategy createProcessorStrategy(RunType runType) {
     switch(runType) {
        ....
     }
   }
   
}
```

Reply:
You did similar thing in `RunIndexWrapperFactory `.